### DISCUSSION TITLE
How do I authenticate to my Azure OpenAI service with Azure AD (/Entra ID)?

### DISCUSSION THREAD

**Discussion by marcussimonsen78** [microsoft/aisdkforsapabap#29] [Q&A]
When I look through the documentation [here](https://microsoft.github.io/aisdkforsapabap/docs/Initial%20Setup/azure-open-ai/azoai-key-endpoint) it shows how to authenticate using access keys. But we have given users access via Azure AD/Entra ID but in the SDK it still prompts for an access key? Is this process described anywhere?

**Comment by gopalnair** [comment]
Hi @marcussimonsen78 ,
   The use of AAD token is mentioned in the following documentation page: (Initialize SDK) 
https://microsoft.github.io/aisdkforsapabap/docs/working-with-sdk/initialize_sdk#api_type


Here, we specify the api type parameter during initialization, and - instead of the key, we pass in the bearer token.

![image](https://github.com/microsoft/aisdkforsapabap/assets/1298551/14095344-b712-4271-9ae2-c88317ac9825)


While the information is available in the page, it seems like, its something that we should showcase more - perhaps just under the key based mechanism. I will work towards updating the documentation website around this.

Please let me know if the issue is resolved and you are able to use the AI SDK with AAD token.

Best Regards,
Gopal Nair.
